packages/remix-react/links.ts

```diff
@@ -425,7 +425,11 @@ export function getDataLinkHrefs(
   let path = parsePathPatch(page);
   return dedupeHrefs(
     matches
-      .filter((match) => manifest.routes[match.route.id].hasLoader)
+      .filter(
+        (match) =>
+          manifest.routes[match.route.id].hasLoader &&
+          !manifest.routes[match.route.id].hasClientLoader
+      )
```

Comment:
Check with @ryanflorence on this - current behavior is desired for folks who are calling `await serverLoader()` unconditionally because we can prefetch that so it resolves immediately.  This new solution hurts those scenarios in favor of apps that conditionally call `serverLoader()`.  Maybe we need to provide some way to enable "prefetching" (pre-executing) of client loaders - either by stashing their results into an internal cache or adding the `type` field to loaders we've previously discussed.

Reply:
👍 Chatted and agreed we should only prefetch module links when a client loader exists

Reply:
Would be great to have the option to prefetch/pre-execute client loaders. In my case I am using a library that has its own cache, so calling the client loader is all that's needed.